temporal-spring-boot-autoconfigure-alpha/src/main/java/io/temporal/spring/boot/autoconfigure/template/WorkersTemplate.java

```diff
@@ -115,90 +114,126 @@ WorkerFactory createWorkerFactory(WorkflowClient workflowClient) {
-                Worker worker = workerFactory.tryGetWorker(taskQueue);
-                if (worker == null) {
-                  log.info(
-                      "Creating a worker with default settings for a task queue '{}' "
-                          + "caused by an auto-discovered activity class {}",
-                      taskQueue,
-                      targetClass);
-                  worker = workerFactory.newWorker(taskQueue);
-                  workers.add(worker);
-                }
-
-                try {
-                  worker.registerActivitiesImplementations(bean);
-                  log.info(
-                      "Registering auto-discovered activity bean '{}' of class {} on task queue '{}'",
-                      beanName,
-                      targetClass,
-                      taskQueue);
-                } catch (TypeAlreadyRegisteredException registeredEx) {
-                  log.info(
-                      "Skipping auto-discovered activity bean '{}' for task queue '{}' "
-                          + "as activity type '{}' is already registered on the worker",
-                      beanName,
-                      taskQueue,
-                      registeredEx.getRegisteredTypeName());
-                }
+  private void configureActivityBeansByTaskQueue(
+      WorkerFactory workerFactory,
+      Workers workers,
+      Map<String, Object> autoDiscoveredActivityBeans) {
+    autoDiscoveredActivityBeans.forEach(
+        (beanName, bean) -> {
+          Class<?> targetClass = AopUtils.getTargetClass(bean);
+          ActivityImpl annotation = AnnotationUtils.findAnnotation(targetClass, ActivityImpl.class);
+          if (annotation != null) {
+            for (String taskQueue : annotation.taskQueues()) {
+              Worker worker = workerFactory.tryGetWorker(taskQueue);
+              if (worker == null) {
+                log.info(
+                    "Creating a worker with default settings for a task queue '{}' "
+                        + "caused by an auto-discovered activity class {}",
+                    taskQueue,
+                    targetClass);
+                worker = workerFactory.newWorker(taskQueue);
+                workers.addWorker(new WorkerDescriptor(worker, null));
               }
+
+              configureActivityImplementationAutoDiscovery(
+                  worker, bean, beanName, targetClass, null);
             }
-          });
+          }
+        });
+  }
+
+  private void configureWorkflowImplementationsByWorkerName(
+      Workers workers, Collection<Class<?>> autoDiscoveredWorkflowImplementationClasses) {
+    for (Class<?> clazz : autoDiscoveredWorkflowImplementationClasses) {
+      WorkflowImpl annotation = clazz.getAnnotation(WorkflowImpl.class);
+
+      for (String workerName : annotation.workers()) {
+        Worker worker = workers.getByName(workerName);
+        if (worker == null) {
+          throw new BeanDefinitionValidationException(
```

Comment:
I almost think we should encourage this indirection just because I find the "worker with name and task queue required" exception to be clearer than creating a worker with default settings when you don't find one for a task queue (you won't know a mistyped task queue name is in your annotation until too late).

Reply:
I think that default workers are mostly "quick project to try things" type of deal. Production users will have a config section. But I think it's fine to make workers auto-discoverable for people who want things up and running as easy as possible.